components/report/reportContents.tsx

```diff
@@ -8,16 +8,40 @@ import ChartCard from 'components/report/chart-card';
 import { BankAccount } from 'integrations/stitch/types';
 import TopExpensesCard from './top-expenses-card';
 import TopDebitOrderCard from 'components/report/top-debit-order-card';
+import AccountsCard from './bank-accounts';
 
-export function ReportContents(props: { bankAccount: BankAccount }): JSX.Element {
+export function ReportContents(props: { bankAccounts: BankAccount[]}): JSX.Element {
+    const bankAccounts = props.bankAccounts.filter(x => x.accountType === 'cheque' || x.accountType === 'current') ?? props.bankAccounts.find(x => x.accountType === 'savings');
```

Comment:
`filter` always returns an array, so we'll never hit the savings case here. Let's rather use a few lines to make the intention clearer:

```suggestion
    const chequeOrCurrentAccounts = props.bankAccounts.filter(x => x.accountType === 'cheque' || x.accountType === 'current');
    const savingsAccounts = props.bankAccounts.filter(x => x.accountType === 'savings');
    const bankAccounts = chequeOrCurrentAccounts.length !== 0 ? chequeOrCurrentAccounts : savingsAccounts;
```

This also fixes the savings case where `find` was returning a single account instead of a list, which would've broken `sort` on the next line.